skrub/_reporting/_table_report.py

```diff
@@ -134,31 +140,38 @@ def __init__(
         self.dataframe = dataframe
         self.verbose = verbose
         self.max_plot_columns = max_plot_columns
+        self.max_association_columns = max_association_columns
+        self.n_columns = sbd.shape(self.dataframe)[1]
+
+        if self.max_plot_columns is None:
+            self.max_plot_columns = self.n_columns
+        if self.max_association_columns is None:
+            self.max_association_columns = self.n_columns
 
     def __repr__(self):
         return f"<{self.__class__.__name__}: use .open() to display>"
 
     @functools.cached_property
-    def _summary_with_plots(self):
-        return summarize_dataframe(
-            self.dataframe, with_plots=True, title=self.title, **self._summary_kwargs
-        )
+    def _lightify_summary(self):
```

Comment:
the reason why we had 2 cached properties is that the json does not need the plots so we might want either a full summary or a plot-less one, and both would be cached.

Now if we don't have just 2 kinds of report but several parameters we should just give up on that it wasn't really necessary, and more a relic from an early version where the tablereport was also able to produce output in text format.

We can have just

```
@functools.cached_property
def _summary(self):
    n_columns = sbd.shape(self.dataframe)[1]
    with_plots = self.max_plot_columns is None or self.max_plot_columns >= n_columns
    with_associations = ...
    return summarize_dataframe(self.dataframe, with_plots=with_plots, ...)
```

(and no more `_lightify_summary` nor `_get_summary` )